src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -11,20 +20,55 @@ namespace Util
         return vm;
     }
 
-    void VisualizerMessenger::initializePublisher(ros::NodeHandle node_handle)
+    void VisualizerMessenger::onConnection()
     {
-        if (this->publisher)
+        // The io_context is required for all I/O
+        boost::asio::io_service ioc{1};
+
+        // The acceptor receives incoming connections
+        // TODO: make IP and port a configurable constant somewhere?
```

Comment:
I think putting them in the `constants.h` file would be fine here

Reply:
Done